packages/pluggableWidgets/accordion-web/src/components/AccordionGroup.tsx

```diff
@@ -1,65 +1,119 @@
+                }, 50);
+            } else {
+                rootElement.current.classList.add("widget-accordion-group-expanding");
+                rootElement.current.classList.remove("widget-accordion-group-collapsed");
+
+                setTimeout(() => {
+                    if (contentWrapperElement.current && contentElement.current) {
+                        setContentWrapperHeightToContentHeight(contentWrapperElement.current, contentElement.current);
+                    }
+                }, 50);
+            }
+        } else if (props.collapsed !== previousCollapsedPropValue && (!animateCollapsing || !props.visible)) {
+            setPreviousCollapsedPropValue(props.collapsed);
         }
-    }, [group, accordionGroupsDispatch]);
+    }, [props.collapsed, props.visible, previousCollapsedPropValue, animateCollapsing]);
 
-    if (!group.visible) {
+    if (!props.visible) {
         return null;
     }
 
     return (
         <section
+            ref={rootElement}
             className={classNames(
                 "widget-accordion-group",
-                { "widget-accordion-group-collapsed": group.collapsed },
-                group.dynamicClassName
+                {
+                    "widget-accordion-group-collapsed": previousCollapsedPropValue
+                },
+                props.dynamicClassName
             )}
         >
             <header
                 className={classNames("widget-accordion-group-header", {
-                    "widget-accordion-group-header-clickable": accordionGroupsDispatch
+                    "widget-accordion-group-header-clickable": props.toggleCollapsed,
+                    "widget-accordion-group-header-icon-left": props.toggleCollapsed && showHeaderIcon === "left",
+                    "widget-accordion-group-header-icon-right": props.toggleCollapsed && showHeaderIcon === "right"
                 })}
-                onClick={accordionGroupsDispatch ? toggleContentVisibility : undefined}
+                onClick={props.toggleCollapsed}
             >
-                {group.header}
+                {props.header}
+                {props.toggleCollapsed && (showHeaderIcon === "left" || showHeaderIcon === "right")
+                    ? props.generateIcon?.(previousCollapsedPropValue ?? false)
+                    : null}
             </header>
-            <div className={"widget-accordion-group-content"}>{divContentMounted ? group.content : undefined}</div>
+            <div
+                ref={contentWrapperElement}
+                className={"widget-accordion-group-content-wrapper"}
+                onTransitionEnd={completeCollapsing}
+            >
+                <div ref={contentElement} className={"widget-accordion-group-content"}>
+                    {props.content}
+                </div>
+            </div>
         </section>
     );
 }
+
+function setContentWrapperHeightToContentHeight(contentWrapper: HTMLDivElement, content: HTMLDivElement): void {
```

Comment:
You can be more specific in this function.. like receive the target height and the target element and rename the method to `setHeight`